elasticapm/transport/base.py

```diff
@@ -176,11 +176,11 @@ def _process_queue(self):
             if flush:
                 if buffer_written:
                     self._flush(buffer, forced_flush=forced_flush)
-                elif forced_flush and "/localhost:" in self.client.config.server_url:
+                elif forced_flush and any(x in self.client.config.server_url for x in ("/localhost:", "/127.0.0.1:")):
                     # No data on buffer, but due to manual flush we should send
                     # an empty payload with flushed=true query param, but only
                     # to a local APM server (or lambda extension)
-                    self.send(None, flushed=True)
+                    self.send({}, forced_flush=True)
```

Comment:
That `Transport.send()` passes `data` to `self.http.urlopen(..., body=data, ...)`, which is https://urllib3.readthedocs.io/en/stable/reference/urllib3.connectionpool.html#urllib3.HTTPConnectionPool.urlopen which defines:

> body – Data to send in the request body, either [str](https://docs.python.org/3/library/stdtypes.html#str), [bytes](https://docs.python.org/3/library/stdtypes.html#bytes), an iterable of [str](https://docs.python.org/3/library/stdtypes.html#str)/[bytes](https://docs.python.org/3/library/stdtypes.html#bytes), or a file-like object.

So would `b''` or `''` be more appropriate than `{}` for an empty value?

Reply:
Good catch, thanks.